### DISCUSSION TITLE
Add cancel download/import button at osu!direct overlay

### DISCUSSION THREAD

**Discussion by LanTenDragon** [ppy/osu#13055] [Ideas]
**Describe the new feature:** In the osu!direct overlay, the download button should change to a cancel button after the player has started a download. 

This would be a quality of life improvement, as sometimes when I change my mind on downloading a map, I have to reach the notification overlay, and by that time, the import is already completed.   

**Proposal designs of the feature:** The download button changes to red and and shows an "X". Text on hover is "Cancel"

**Comment by peppy** [comment]
Maybe a dialog on clicking the in-progress button to allow cancelling? Not sure.

**Comment by elleyer** [comment]
I'm not sure it's an important thing because typical beatmap uses only few MB's of the memory, and you can delete it at any moment. Also it downloads in a few seconds, so you will not have a time for cancel it.

**Comment by Gabixel** [comment]
That's true, but this feature can come in handy for some rare moments: the easiest one is the misclicking (but that's just an example). My idea is to double-click on the download button (or just pressing a cancel button next to the download one).

**Comment by RoARene317** [comment]
In Stable version there isn't an option to cancel download. Unless in Lazer there is a download manager that control the download. Since before download , people being played the replay and right now you can see the beatmap listings without opening the browser. So you can consider which map and who mapped the beatmap.

**Comment by peppy** [comment]
In stable there *is* an option. In lazer you *can* already cancel the download via the notification area.

**Comment by frenzibyte** [comment]
Are we agreeing on a design here?

**Comment by peppy** [comment]
A dialog popup allowing cancelling would be a good start, as I mentioned above.

**Reply by Spooghetti420** [reply]
I was about to write a proposal for this exact feature. Even today, I still think it would be beneficial to have a clear "X" symbol to replace the download button on the beatmap listing. @peppy, what would the dialog popup entail? At any rate, I am interested in trying to implement this feature if it is still being considered.

**Reply by peppy** [reply]
You could try using a `Popover` which appears when you click the download progress display, and contains a cancel button. See the multiplayer password entry popover for ideas.
